### PR TITLE
[APM-CI][Integration Tests][GO] Move to Go modules to fix versions

### PR DESCRIPTION
**Pull request by v1v**
Closes https://github.com/elastic/apm-integration-testing/issues/434

## Highlights
- Move to Go latest to be able to use Go modules as suggested by @axw 
- Set modules based on the ones defined in https://github.com/elastic/apm-agent-go/blob/master/go.mod

## Test cases
- Build docker images

```bash
$ python scripts/compose.py start 7.0.0   --no-apm-server-dashboards --no-apm-server-self-instrument --no-kibana --with-agent-go-net-http --with-opbeans-go --force-build
....

Successfully tagged apm-integration-testing_opbeans-go:latest
Pulling elasticsearch          ... done
Pulling apm-server             ... done
Pulling postgres               ... done
Pulling redis                  ... done
Pulling opbeans-load-generator ... done
localtesting_7.0.0_elasticsearch is up-to-date
Creating localtesting_7.0.0_redis ... 
Creating localtesting_7.0.0_redis    ... done
Creating localtesting_7.0.0_postgres ... done
Recreating gonethttpapp              ... done
Creating localtesting_7.0.0_opbeans-go ... done
Creating localtesting_7.0.0_opbeans-load-generator ... done

```

- Run ITs
```bash
$ make docker-test-agent-go

tests/agent/test_go.py::test_concurrent_req_go_nethttp_foobar {'name': 'gonethttpapp', 'runtime': {'name': 'gc', 'version': 'go1.12.5'}, 'language': {'name': 'go', 'version': 'go1.12.5'}}
{'name': 'gonethttpapp', 'runtime': {'name': 'gc', 'version': 'go1.12.5'}, 'language': {'name': 'go', 'version': 'go1.12.5'}}
PASSED     [ 25%]
tests/agent/test_go.py::test_concurrent_req_go_nethttp {'name': 'gonethttpapp', 'runtime': {'name': 'gc', 'version': 'go1.12.5'}, 'language': {'name': 'go', 'version': 'go1.12.5'}}
{'name': 'gonethttpapp', 'runtime': {'name': 'gc', 'version': 'go1.12.5'}, 'language': {'name': 'go', 'version': 'go1.12.5'}}
PASSED            [ 50%]
tests/agent/test_go.py::test_req_go_nethttp PASSED                       [ 75%]
tests/agent/test_go.py::test_go_nethttp_error PASSED                     [100%]

```

### CODE REVIEW

File: docker/go/nethttp/go.mod [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	github.com/santhosh-tekuri/jsonschema v1.2.4 // indirect
+	github.com/stretchr/objx v0.2.0 // indirect
+	go.elastic.co/apm v1.3.0
+	go.elastic.co/apm/module/apmhttp v1.3.0
+	golang.org/x/crypto v0.0.0-20190426145343-a29dc8fdc734 // indirect
+	golang.org/x/net v0.0.0-20190503192946-f4e77d36d62c // indirect
+	golang.org/x/text v0.3.2 // indirect
+	golang.org/x/tools v0.0.0-20190503185657-3b6f9c0030f7 // indirect
+	gopkg.in/yaml.v2 v2.2.2 // indirect
+)
```
Comment (axw) [inline docker/go/nethttp/go.mod]: There need to be a couple of "replace" lines like this:

https://github.com/elastic/apm-agent-go/blob/47cc289609d7000794e858f84652c5ddbe17e623/module/apmecho/go.mod#L17-L19

They should point to the location where the apm-agent-go repo is checked out. Without those, `go build` will fetch `go.elastic.co/apm` from GitHub.

File: docker/go/nethttp/Dockerfile [hunk 2]
```diff
@@ -1,13 +1,14 @@
-FROM golang:1.10
+FROM golang:1.11
```
Comment (axw) [inline docker/go/nethttp/Dockerfile]: Maybe we should just use `golang:latest`? Or at least `golang:1.12`.

**Comment by axw** [comment]
@v1v I've created https://github.com/v1v/apm-integration-testing/pull/2 which addresses the issues I noted. Feel free to pull those changes into your branch if they look good to you.

**Comment by v1v** [comment]
Jenkins run the Tests PlEaSe

**Comment by v1v** [comment]
jenkins run the tests

**Comment by v1v** [comment]
jenkins run the tests please

**Comment by kuisathaverat** [comment]
Do we need the `docker/go/nethttp/go.sum` file?

**Pull request merged**